tests/integration/any_charm.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""This code snippet is used to be loaded into any-charm which is used for integration tests."""
+import os
+import pathlib
+import signal
+import subprocess
+
+from any_charm_base import AnyCharmBase
+from ingress import IngressRequires
+
+
+class AnyCharm(AnyCharmBase):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.ingress = IngressRequires(
+            self,
+            {"service-hostname": self.app.name, "service-name": self.app.name, "service-port": 80},
+        )
+
+    def update_ingress(self, ingress_config):
+        self.ingress.update_config(ingress_config)
+
+    @staticmethod
+    def start_server(port: int = 80):
+        """Start an HTTP server daemon."""
+        www_dir = pathlib.Path("/tmp/www")
+        www_dir.mkdir(exist_ok=True)
```

Comment:
This could fail if the directory already exists *and* we don't have the rights to create the "ok" file in it.  
But I hardly see a scenario where this would be an issue in our own tests

Reply:
Couldn't we use "tempfile" ? https://docs.python.org/3/library/tempfile.html

Reply:
To be honest, we actually don't even need a file in `/tmp/www` for the test to work (we just need a valid response from the webserver).  
I could remove the `ok` file for now.  

Later, if we want to test the content-cache operator with specific payloads from the webserver we will need a known path to the served file and creating a `NamedTemporaryFile` will not be an option

Reply:
So should this part be removed then and we can delay a conversation on whether to use a temporary directory or the `www` directory until then? Or do you think it would be better to have the directory? It also looks like the `www_dir` is passed to the server start command anyway, so I'm guessing any directory would be ok even a `NamedTemporaryFile`? Or am I missing something?

Reply:
There is also `TemporaryDirectory` if a directory is needed

Reply:
I used `mkdtemp` to create www_dir and removed the "ok" file